### PR TITLE
🚧  Implement 2 to master (v0.2.0)

### LINKED ISSUE #51: The response of the authentication should contain information about the anonymous status
**Is your feature request related to a problem? Please describe.**
Currently when calling `session.begin()` it is not clear wether you are logged in as an anonymous user or not.
The response contains the username and it would be possible to check if the username is the same as the anonymous user, but this would definitely create a large potential for error.

Let's look at an example:
The change of the anonymous username in `v0.2.0` would be a breaking change cause every user of the snek-client had to adjust there codebase on the new username.

**Describe the solution you'd like**
It would be a lot cooler if the response of `session.begin()` would contain a boolean, let's call it `anonymous` which tells if I got authenticated as anonymous or not.

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the `session.begin()` response does not include a anonymous status. #51 
- Currently the client does not apply with the newest engine version. #62

**What is the new behavior (if this is a feature change)?**
- Now the `session.begin()` response does include an anonymous status. Closes #51 
- Now the client applies with the newest engine release. #Closes 62

**Other information**:
- 🐍

### CODE REVIEW

**Review by Aichnerc — APPROVED** [review]
LGTM

**Review by pinterid — APPROVED** [review]
LGTM

File: src/config.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "anonUser": {
+    "username": "cisco",
+    "password": "ciscocisco"
+  }
+}
```
Comment (kleberbaum) [inline src/config.json:6]:
```suggestion
}

```

Comment (kleberbaum) [inline src/config.json:6]: I think a YAML file would be 100 times cooler. 💯

File: src/templates/snek/gql/mutations/user.ts [hunk 2]
```diff
@@ -17,17 +17,16 @@ import gql from "graphql-tag";
...
-      values: $values
-    ) {
-      result
-      errors {
-        name
-        errors
+    registration: registrationFormPage(
+      url: "/registration/",
+      token: $token,
+      values: $values) {
```
Comment (kleberbaum) [inline src/templates/snek/gql/mutations/user.ts]:
```suggestion
      values: $values
    ) {
```

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge.

File: src/config.json (hunk 1 above)
Comment (schettn) [reply src/config.json:6]: In contrast to YAML, JSON can be imported as follows:
```
//> Config
import Config from "../../../config.json";
```
This notation guarantees full IntelliSense.

---

A method which works with YAML and JSON is as follows:
```
var config = require("../../../config.json")
```

But this notation does not have any IntelliSense support without implementing addidional interfaces.

Due to the small scope of the configuration this change would results in more work than benefits.

**Pull request closed without merging**